src/poker.js

```diff
@@ -1,10 +1,85 @@
-function winningPair() {}
+const cardMap = {
+  3: 3,
+  4: 4,
+  5: 5,
+  6: 6,
+  7: 7,
+  8: 8,
+  9: 9,
+  10: 10,
+  J: 11,
+  Q: 12,
+  K: 13,
+  A: 14
+}
+
+function winningPair(pairOne, pairTwo) {
+  if (pairOne[0] !== pairOne[1] && pairTwo[0] !== pairTwo[1]) return []
+
+  if (pairOne[0] === pairOne[1] && pairTwo[0] === pairTwo[1]) {
+    if (cardMap[pairOne[0]] > cardMap[pairTwo[0]]) return pairOne
+    return pairTwo
+  }
+
+  if (pairOne[0] === pairOne[1]) return pairOne
+  return pairTwo
+}
 
 // Extension criteria
 
-function winningPairFromArray() {}
+function winningPairFromArray(pairArray) {
+  const resultArray = []
+
+  for (const i in pairArray) {
+    if (pairArray[i][0] === pairArray[i][1]) resultArray.push(pairArray[i][0])
+  }
+
+  if (resultArray.length === 0) return []
+
+  let highestString = '3'
+  for (const i in resultArray) {
+    if (cardMap[resultArray[i]] > cardMap[highestString])
+      highestString = resultArray[i]
+  }
+
+  return [highestString, highestString]
+}
+
+function getHighest(resultArray) {
+  let highestString = '3'
+  for (const i in resultArray) {
+    if (cardMap[resultArray[i]] > cardMap[highestString])
+      highestString = resultArray[i]
+  }
+  return highestString
+}
+
+function winning3CardHand(pairArray) {
+  const resultArray = []
+  const result3Array = []
+
+  for (const i in pairArray) {
+    if (
+      pairArray[i].length === 3 &&
+      pairArray[i][0] === pairArray[i][1] &&
+      pairArray[i][0] === pairArray[i][2]
+    ) {
```

Comment:
I would extract `pairArray[i]` into a variable of its own so you don't have it in so many places:

```js
const hand = pairArray[i]

if (hand.length === 3 && hand[0] === hand[1]...)
```

You could go one step further and destructure the array into their own variables so make things even easier to read:

```js
const hand = pairArray[i]
const [one, two, three] = hand

if (hand.length === 3 && one === two && one === three) {
```